src/context/Queue.ts

```diff
@@ -18,6 +18,11 @@ export enum ActionType {
     Signup = 'Signup',
 }
 
+export interface Metadata {
+    id?: string
+    transaction?: string
```

Comment:
Maybe make this `transactionId` to make it more obvious it's a string and not an object?